src/effects/throw_spiny.c

```diff
@@ -1,14 +1,171 @@
 #include "common.h"
 #include "effects_internal.h"
+#include "effects.h"
 
 void throw_spiny_appendGfx(void* effect);
 
-INCLUDE_ASM(s32, "effects/throw_spiny", throw_spiny_main);
+void throw_spiny_init(void);
+void throw_spiny_update(EffectInstance* effect);
+void throw_spiny_render(EffectInstance* effect);
+
+extern Gfx D_09000800[];
+extern Gfx D_090008D8[];
+extern Gfx D_090009F0[];
+
+Gfx* D_E00C8710[2] = { D_09000800, D_090008D8 };
+
+u8 D_E00C8718[8] = { 110, 150, 130, 110, 100, 95, 100, 0 };
+u8 D_E00C8720[8] = { 80, 60, 80, 100, 120, 110, 100, 0 };
+
+typedef struct ThrowSpinyFXData {
+/* 0x00 */ s32 unk_00;
+/* 0x04 */ Vec3f pos;
+/* 0x10 */ f32 unk_10;
+/* 0x14 */ f32 unk_14;
+/* 0x18 */ f32 unk_18;
+/* 0x1C */ f32 unk_1C;
+/* 0x20 */ f32 unk_20;
+/* 0x24 */ f32 unk_24;
+/* 0x28 */ s32 life;
+/* 0x2C */ s32 lifeDuration;
+/* 0x30 */ s32 unk_30;
+/* 0x34 */ s32 unk_34;
+/* 0x38 */ s32 unk_38;
+/* 0x3C */ s32 rgba;
+/* 0x40 */ f32 unk_40;
+/* 0x44 */ f32 unk_44;
+/* 0x48 */ f32 gravity;
+/* 0x4C */ f32 unk_4C;
+/* 0x50 */ f32 yaw;
+/* 0x54 */ f32 rotationSpeed;
+/* 0x58 */ f32 xScale;
+/* 0x5C */ f32 yScale;
+/* 0x60 */ u32 state;
+/* 0x64 */ s32 unk_64;
+/* 0x68 */ s32 timeUntilFall; //how quickly should spiny fall to ground ?
+} ThrowSpinyFXData; //sizeof 0x6C
+
+//during spiny surge
+EffectInstance* throw_spiny_main(s32 arg0, f32 arg1, f32 arg2, f32 arg3, f32 arg4, f32 arg5, f32 arg6, f32 arg7, s32 time) {
```

Comment:
This func needs to be cleaned up. a few basic notes:
* the `int` should be an `s32`
* the functions you put into the bp fields don't need casting or parens or the & infront
* These _main funcs all look the same at the beginning, so you should take another one that's matched currently and change this to look like that for consistency imo - that'll solve the two points above as well. your new_var is usually "numParts" iirc, etc etc